lib/cext/preprocess.rb

```diff
@@ -11,6 +11,26 @@
 
 ALLOCA_LOCALS = /^(\s+)VALUE\s*\*\s*([a-z_][a-zA-Z_0-9]*)\s*=\s*(\(\s*VALUE\s*\*\s*\)\s*)?alloca\(/
 
+# Found in nokogiri
+XML_NODE_SET_PATCH = <<-EOF
+  switch (rb_tr_to_int_const(rb_range_beg_len(arg, &beg, &len, (long)node_set->nodeNr, 0))) {
+  case Qfalse_int_const:
+    break;
+  case Qnil_int_const:
+EOF
+
+# Found in pg
+PG_BINARY_ENCODER_PATCH = <<-EOF	
+	switch(rb_tr_to_int_const(value)){
+		case Qtrue_int_const : mybool = 1; break;
+		case Qfalse_int_const : mybool = 0; break;
+EOF
+
+PATCHED_FILES = {'xml_node_set.c'      => {:match =>  /[[:blank:]]*switch\s*\(.*Qnil:/m, 
+                                           :replacement => XML_NODE_SET_PATCH},
+                 'pg_binary_encoder.c' => {:match => /[[:blank:]]*switch\s*\(.*Qfalse.*break;/m,
+                                           :replacement => PG_BINARY_ENCODER_PATCH}}
```

Comment:
Is `[[:blank:]]` needed over `\s`, what's the difference?
It seems better to match non-greedy here, with `.*?`, and make sure there is a `:` after `Qfalse`.